src/views/studio/CreateMemberView/CreateMemberView.tsx

```diff
@@ -1,62 +1,109 @@
 import { ActionDialog } from '@/components/Dialogs'
 import { absoluteRoutes } from '@/config/routes'
+import { useActiveUser, useConnectionStatus } from '@/hooks'
 import { Spinner, Text, TextField } from '@/shared/components'
 import TextArea from '@/shared/components/TextArea'
 import { textFieldValidation, urlValidation } from '@/utils/formValidationOptions'
 import { debounce } from 'lodash'
 import React, { useEffect, useState } from 'react'
 import { useForm } from 'react-hook-form'
 import { useNavigate } from 'react-router'
+import { FAUCET_URL } from '@/config/urls'
 import { Form, StyledButton, Wrapper, StyledText, Header, Hero, SubTitle, StyledAvatar } from './CreateMemberView.style'
+import { useMemberships } from '@/api/hooks'
+import MessageDialog from '@/components/Dialogs/MessageDialog'
 
 type Inputs = {
   handle: string
-  avatarUri: string
+  avatar: string
   about: string
 }
 
 const CreateMemberView = () => {
-  const [loading, setLoading] = useState(false)
-  const [avatarImageUrl, setAvatarImageUrl] = useState('')
+  const { activeUser } = useActiveUser()
+  const { nodeConnectionStatus } = useConnectionStatus()
+
   const navigate = useNavigate()
-  const { register, handleSubmit, errors, trigger, setError } = useForm<Inputs>({
+  const { register, handleSubmit, errors, trigger, setError: setInputError } = useForm<Inputs>({
     shouldFocusError: false,
     defaultValues: {
       handle: '',
-      avatarUri: '',
+      avatar: '',
       about: '',
     },
   })
 
-  useEffect(() => {
-    if (!loading) {
-      return
-    }
-    const timeout = setTimeout(() => {
-      setLoading(false)
-      navigate(absoluteRoutes.studio.signIn())
-    }, 3000)
-    return () => {
-      clearTimeout(timeout)
+  const [shouldFetchMemberships, setShouldFetchMemberships] = useState(false)
+  const [error, setError] = useState<string | undefined>()
+  const [avatarImageUrl, setAvatarImageUrl] = useState('')
+  const [isLoadingDialogVisible, setIsLoadingDialogVisible] = useState(false)
+
+  const { memberships, startPolling } = useMemberships(
+    {
+      where: {
+        controllerAccount_in: [activeUser.accountId || ''],
```

Comment:
Why not `controllerAcount_eq`?

Reply:
if I change that it just won't redirect the user to the `/signin` after creating a member. It will go back to `/sign/join`

Reply:
Right, because then this would a different query than the one used for route auth so it wouldn't be refetched automatically. This is tricky, but let's keep it then. We should add a comment here to explain what's happening though